graphgym/contrib/config/roland.py

```diff
@@ -0,0 +1,196 @@
+    # are for validation and the last 10% snapshots are for testing.
+    cfg.dataset.split_method = 'default'
+
+    # In the case of live-update, whether to predict all edges at time t+1.
+    cfg.dataset.link_pred_all_edges = False
+    # ----------------------------------------------------------------------- #
+    # Customized options: `transaction` for ROLAND dynamic graphs.
+    # ----------------------------------------------------------------------- #
+
+    # example argument group
+    cfg.transaction = CN()
+
+    # whether use snapshot
+    cfg.transaction.snapshot = False
+
+    # snapshot split method 1: number of snapshots
+    # split dataset into fixed number of snapshots.
+    cfg.transaction.snapshot_num = 100
+
+    # snapshot split method 2: snapshot frequency
+    # e.g., one snapshot contains transactions within 1 day.
+    cfg.transaction.snapshot_freq = 'D'
+
+    cfg.transaction.check_snapshot = False
+
+    # how to use transaction history
+    # full or rolling
+    cfg.transaction.history = 'full'
+
+    # type of loss: supervised / meta
+    cfg.transaction.loss = 'meta'
+
+    # feature dim for int edge features
+    cfg.transaction.feature_int_dim = 32
+    cfg.transaction.feature_edge_int_num = [50, 8, 252, 252, 3, 3]
+    cfg.transaction.feature_node_int_num = [0]
+
+    # feature dim for amount (float) edge feature
+    cfg.transaction.feature_amount_dim = 64
+
+    # feature dim for time (float) edge feature
+    cfg.transaction.feature_time_dim = 64
+
+    #
+    cfg.transaction.node_feature = 'raw'
+
+    # how many days look into the future
+    cfg.transaction.horizon = 1
+
+    # prediction mode for the task; 'before' or 'after'
+    cfg.transaction.pred_mode = 'before'
+
+    # number of periods to be captured.
+    # set to a list of integers if wish to use pre-defined periodicity.
+    # e.g., [1,7,28,31,...] etc.
+    cfg.transaction.time_enc_periods = [1]
+
+    # if 'enc_before_diff': attention weight = diff(enc(t1), enc(t2))
+    # if 'diff_before_enc': attention weight = enc(t1 - t2)
+    cfg.transaction.time_enc_mode = 'enc_before_diff'
+
+    # how to compute the keep ratio while updating the recurrent GNN.
+    # the update ratio (for each node) is a function of its degree in [0, t)
+    # and its degree in snapshot t.
+    cfg.transaction.keep_ratio = 'linear'
+
+    # ----------------------------------------------------------------------- #
+    # Customized options: metrics.
+    # ----------------------------------------------------------------------- #
+
+    cfg.metric = CN()
+    # How many negative edges for each node to compute rank-based evaluation
+    # metrics such as MRR and recall at K.
+    # E.g., if multiplier = 1000 and a node has 3 positive edges, then we
+    # compute the MRR using 1000 randomly generated negative edges
+    # + 3 existing positive edges.
+    # Use 100 ~ 1000 for fast and reliable results.
+    cfg.metric.mrr_num_negative_edges = 1000
```

Comment:
How can we set negative edges as all the edges?

Reply:
We can set `cfg.metric.mrr_method = 'all'` to forcely enable it, but typically we don't use all negative edges since it's super slow.